helix-zk/src/main/java/org/apache/helix/zkscale/zk/PathBasedZkSerializer.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.helix.zkscale.zk;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.I0Itec.zkclient.exception.ZkMarshallingError;
+
+public interface PathBasedZkSerializer {
```

Comment:
We need a separate folder path API to add these kind of interfaces.

Reply:
+1 on this, since we are making  a new  module, let us make it clearly organized that user should be very easily  to tell which  classes/interfaces are public API, which are our internal implementations (they are not supposed to use them directly).